manifests/resource/upstream.pp

```diff
@@ -18,82 +31,116 @@
 # Sample Usage:
 #  nginx::resource::upstream { 'proxypass':
 #    ensure  => present,
-#    members => [
-#      'localhost:3000',
-#      'localhost:3001',
-#      'localhost:3002',
-#    ],
+#    members => {
+#      'localhost:3001' => {
+#        server => 'localhost',
+#        port   => 3000,
+#      },
+#      'localhost:3002' => {
+#        server => 'localhost',
+#        port   => 3002,
+#      },
+#      'localhost:3003' => {
+#        server => 'localhost',
+#        port   => 3003,
+#      },
+#    },
 #  }
 #
 #  Custom config example to use ip_hash, and 20 keepalive connections
 #  create a hash with any extra custom config you want.
-#  $my_config = {
-#    'ip_hash'   => '',
-#    'keepalive' => '20',
-#  }
 #  nginx::resource::upstream { 'proxypass':
-#    ensure              => present,
-#    members => [
-#      'localhost:3000',
-#      'localhost:3001',
-#      'localhost:3002',
-#    ],
-#    upstream_cfg_prepend => $my_config,
+#    ensure    => present,
+#    members   => {
+#      'localhost:3001' => {
+#        server => 'localhost',
+#        port   => 3000,
+#      },
+#      'localhost:3002' => {
+#        server => 'localhost',
+#        port   => 3002,
+#      },
+#      'localhost:3003' => {
+#        server => 'localhost',
+#        port   => 3003,
+#      },
+#    },
+#    ip_hash   => true,
+#    keepalive => 20,
 #  }
+#
 define nginx::resource::upstream (
-  Optional[Array] $members                  = undef,
-  $members_tag                              = undef,
-  Enum['present', 'absent'] $ensure         = 'present',
-  Optional[Hash] $upstream_cfg_append       = undef,
-  Optional[Hash] $upstream_cfg_prepend      = undef,
-  $upstream_fail_timeout                    = '10s',
-  $upstream_max_fails                       = undef,
-  Enum['http', 'stream'] $upstream_context  = 'http',
+  Enum['present', 'absent']                 $ensure                 = 'present',
+  Enum['http', 'stream']                    $context                = 'http',
+  Optional[Hash]                            $members                = undef,
+  Optional[String]                          $members_tag            = undef,
+  Optional[Nginx::UpstreamMemberDefaults]   $member_defaults        = undef,
+  Optional[String]                          $hash                   = undef,
+  Optional[Boolean]                         $ip_hash                = undef,
```

Comment:
Please check all parameters with the `Boolean` dataype. If it makes sense, remove the `Optional` and let them default to `True` or `False`

Reply:
You're right, again, I'll fix that, too.